### PR TITLE
Add a GitHub Action to inject preview links in PRs

### PR DESCRIPTION
**Pull request by pradyunsg**
This makes easier to review pull requests by providing the preview links in the PR description.

### CODE REVIEW

**Comment by pradyunsg** [comment]
This is also used on https://github.com/python/peps/, where you can see how this injects the name (eg: https://github.com/python/peps/pull/3522).

**Comment by sinoroc** [comment]
Thanks! I knew I had seen something like this before.

File: .github/workflows/pr-preview-links.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Read the Docs PR preview
+
+on:
+  pull_request_target:
+    types:
+      - opened
```
Comment (chrysle) [inline .github/workflows/pr-preview-links.yml:6]: Should we add this?

```suggestion
    types:
      - opened
      paths:
        - "source/**"
```

Comment (pradyunsg) [reply .github/workflows/pr-preview-links.yml:6]: I considered this, but we have .readthedocs.yml and requirements.txt outside of this directory which do affect the RTD build. After adding those, I figured that the RTD builds are going to happen again anyway and it's easy-enough to see things. 😅

Comment (chrysle) [reply .github/workflows/pr-preview-links.yml:6]: Sounds fair, resolving!

**Review by chrysle — APPROVED** [review]
LGTM, but let's wait for the expert's (@webknjaz's) review.

**Comment by willingc** [comment]
@webknjaz Sorry if I merged prematurely. I missed @chrysle's note.

**Pull request merged**